cupy/math/misc.py

```diff
@@ -1,8 +1,122 @@
+import cupy
+import cupyx
+
 from cupy import core
 from cupy.core import _routines_math as _math
 from cupy.core import fusion
+from cupy.lib import stride_tricks
+
+
+_dot_kernel = core.ReductionKernel(
+    'T x1, T x2',
+    'T y',
+    'x1 * x2',
+    'a + b',
+    'y = a',
+    '0',
+    'dot_product'
+)
 
-# TODO(okuta): Implement convolve
+
+def convolve(a, v, mode='full'):
+    """Returns the discrete, linear convolution of two one-dimensional sequences.
+
+    Args:
+        a (cupy.ndarray): first 1-dimensional input.
+        v (cupy.ndarray): second 1-dimensional input.
+        mode (str, optional): `valid`, `same`, `full`
+
+    Returns:
+        cupy.ndarray: Discrete, linear convolution of a and v.
+
+    .. seealso:: :func:`numpy.convolve`
+
+    """
+    if a.size == 0:
+        raise ValueError('a cannot be empty')
+    if v.size == 0:
+        raise ValueError('v cannot be empty')
+    if v.ndim > 1:
+        raise ValueError('v cannot be multidimensional array')
+    if v.size > a.size:
+        a, v = v, a
+    a = a.ravel()
+    v = v.ravel()
+
+    method = cupyx.scipy.signal.choose_conv_method(a, v, mode)
+    if method == 'direct':
+        _, out = _dot_convolve(a, v, mode)
+    elif method == 'fft':
+        out = _fft_convolve(a, v, mode)
+    else:
+        raise ValueError('Unsupported method')
+    return out
+
+
+def _fft_convolve(a1, a2, mode='full'):
```

Comment:
Default value is not needed here.
```suggestion
def _fft_convolve(a1, a2, mode):
```